packages/typescript-estree/src/convert.ts

```diff
@@ -564,13 +564,33 @@ export class Converter {
   }
 
   private convertJSXNamespaceOrIdentifier(
-    node: ts.Identifier | ts.ThisExpression,
+    node: ts.JsxNamespacedName | ts.Identifier | ts.ThisExpression,
   ): TSESTree.JSXIdentifier | TSESTree.JSXNamespacedName {
+    // TypeScript@5.1 added in ts.JsxNamespacedName directly
+    // We prefer using that if it's relevant for this node type
+    if (node.kind === ts.SyntaxKind.JsxNamespacedName) {
+      const result = this.createNode<TSESTree.JSXNamespacedName>(node, {
+        type: AST_NODE_TYPES.JSXNamespacedName,
+        namespace: this.createNode(node.namespace, {
+          type: AST_NODE_TYPES.JSXIdentifier,
+          name: node.namespace.text,
+        }),
+        name: this.createNode(node.name, {
+          type: AST_NODE_TYPES.JSXIdentifier,
+          name: node.name.text,
+        }),
+      });
+      this.registerTSNodeInNodeMap(node, result);
+      return result;
+    }
```

Comment:
> [TS already supported parsing namespaced attributes](https://typescript-eslint.io/play/#ts=5.0.3&jsx=true&sourceType=module&showAST=es&code=JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcA5FDvmQNwBQduEAdqvAB5wC8cAPACbAAbnAIQIALgBGyKNzgAiAIwK4AegB8QA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA) and we have AST types for it - so this is a no-op for us.

I think this is a bit more efficient of a transform. But I'm a little lost in the using `JSXIdentifier` vs `Identifier` - do we want to normalize one way or another?

Reply:
You updated your comment but to answer your previous question...

> But I'm a little lost in the using JSXIdentifier vs Identifier - do we want to normalize one way or another?

JSX identifier exists because [JSX is specced to allow names with `-`](https://facebook.github.io/jsx/#prod-JSXIdentifier).

EG `<foo-bar />` creates a `JSXIdentifier { name = 'foo-bar' }` - and this name is obviously is not a valid `Identifier`.
So we cannot normalise one way or the other, sadly 😢

Reply:
Yeah it's a good idea to be more explicit if we can considering that now TS has an explicit node for namespacing.

Reply:
Ha yeah I figured part of it out a bit after commenting. This makes sense, thanks!